BossMod/Components/LineOfSightAOE.cs

```diff
@@ -9,6 +11,9 @@ public abstract class GenericLineOfSightAOE(BossModule module, ActionID aid, flo
     public WPos? Origin { get; private set; } // inactive if null
     public List<(WPos Center, float Radius)> Blockers { get; private set; } = [];
     public List<(float Distance, Angle Dir, Angle HalfWidth)> Visibility { get; private set; } = [];
+    public List<AOEInstance> Safezones = [];
+    public List<AOEShape> UnionShapes = [];
+    public List<AOEShape> DifferenceShapes = [];
```

Comment:
Why do we need all this?

Reply:
he is missing all the custom AOEShape logic, it wont work like this. i basically refactored your whole line of sight component to be more accurate, support rectangle aoes etc.

https://github.com/FFXIV-CombatReborn/BossmodReborn/blob/main/BossMod/Components/LineOfSightAOE.cs

Reply:
It works they way I ported it, my porting method is to donthe least framework modification ad possible.